src/utils/validation.js

```diff
@@ -16,6 +17,8 @@ module.exports = ({components, paths, ajvOptions}) => {
 		removeAdditional: false,
 		...ajvOptions
 	});
+	// expose ajv so that developers can cusomize it
+	ajvFile.rawAjv = ajv;
```

Comment:
What do developers want to do for customising~? Is `ajvOptions` enough for the use case?

Reply:
we want to add keywords and customize error messages. This is npm package to customize error message： https://github.com/epoberezkin/ajv-errors, we must pass ajv instance into this package as a parameter.

Reply:
i see~ So the line needs to be added is
```
require('ajv-errors')(ajv /*, {singleError: true} */);
```

Do you think it is better to directly embed https://github.com/epoberezkin/ajv-errors into our lib instead of exposing `ajv`? I think it is better to keep the interfaces of the lib as few as possible. Otherwise, it would be too diverse for the lib.

it would be nice to support https://github.com/epoberezkin/ajv-errors#options also.

Reply:
And one more case: we want to invoke ajv.addKeywords manually to add keywords. As your reply above, are we better to pass keywords we want to add into swagger-ajv as parameter instead of exposing ajv?